applications/ExaquteSandboxApplication/tests/test_divergencefree_refinement_process.py

```diff
@@ -0,0 +1,135 @@
+meshing_is_available = KratosMultiphysics.kratos_utilities.CheckIfApplicationsAvailable("MeshingApplication")
+
+import KratosMultiphysics.KratosUnittest as UnitTest
+
+
+class TimeAveragingProcessTests(UnitTest.TestCase):
+
+    def testDivergenceFreeRefinementProcess(self):
+        if not meshing_is_available:
+            self.skipTest("Missing required application: MeshingApplication")
+
+        self.__CreateModel()
+
+        settings = KratosMultiphysics.Parameters(r'''
+        [
+            {
+                "kratos_module" : "KratosMultiphysics.ExaquteSandboxApplication",
+                "python_module" : "divergencefree_refinement_process",
+                "process_name"  : "DivergenceFreeRefinementProcess",
+                "Parameters" : {
+                    "model_part_name"                       : "test",
+                    "integration_end_point_control_value"   : 5.0,
+                    "integration_start_point_control_value" : 1.0,
+                    "average_settings": {
+                        "model_part_name"                               : "test",
+                        "variables_list"                                : ["DIVERGENCE"],
+                        "averaged_variables_list"                       : ["AVERAGED_DIVERGENCE"],
+                        "time_averaging_container"                      : "ElementalNonHistorical",
+                        "time_averaging_method"                         : "RootMeanSquare",
+                        "integration_start_point_control_variable_name" : "TIME",
+                        "integration_start_point_control_value"         : 2.0
+                    },
+                    "refinement_settings" : {
+                        "refinement_strategy" : "global_tolerance_strategy",
+                        "reference_variable_name" : "AVERAGED_DIVERGENCE",
+                        "minimal_size"                     : 0.001,
+                        "maximal_size"                     : 10.0,
+                        "mean_distribution_strategy":
+                        {
+                            "target_refinement_coefficient"       : 0.9,
+                            "refinement_bound"                    : 2.0,
+                            "reference_norm_name"                 : "VELOCITY_H1_SEMINORM"
+                        },
+                        "maximum_strategy":
+                        {
+                            "target_refinement_coefficient"       : 0.1,
+                            "refinement_coefficient"              : 2.0
+                        },
+                        "global_tolerance_strategy":
+                        {
+                            "global_tolerance"                 : 0.1
+                        }
+                    }
+                }
+            }
+        ]''')
+
+        factory = KratosProcessFactory(self.model)
+        self.process_list = factory.ConstructListOfProcesses(settings)
```

Comment:
You can directly construct the process you are testing to avoid all these `self.process_list` loops.

Reply:
Done.